docs/application/native/guides/text-input/voice-control-manager.md

```diff
@@ -0,0 +1,1154 @@
+  The voice control manager state changes according to the function calls:
+
+  ```c
+  void get_state()
+  {
+      vc_state_e current_state;
+      int ret;
+      ret = vc_mgr_get_state(&current_state);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get the current voice control manager service state using `vc_mgr_get_service_state()`.
+  If the application uses continuous recognition, the voice control service state can be changed from `VC_SERVICE_STATE_PROCESSING` directly to `VC_SERVICE_STATE_RECORDING`:
+
+  ```c
+  void get_service_state()
+  {
+      vc_service_state_e service_state;
+      int ret;
+      ret = vc_mgr_get_service_state(&service_state);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get the supported languages using `vc_mgr_foreach_supported_languages()` that triggers a separate callback for each language.
+  As long as the callback returns `true`, the function continues to loop over the supported languages. `vc_mgr_foreach_supported_languages()` is not used when the voice control service is in the `VC_SERVICE_STATE_NONE` state:
+
+  ```c
+  bool
+  __supported_language_cb(const char* language, void* user_data)
+  {
+      return true; /* To continue to get the next language */
+
+      return false; /* To stop the loop */
+  }
+
+  void
+  get_supported_language()
+  {
+      int ret;
+      ret = vc_mgr_foreach_supported_languages(__supported_language_cb, NULL);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get the current language using `vc_mgr_get_current_language()`.
+  Use the language changed callback to get notifications for any language change. `vc_mgr_get_current_language()` is not used when the voice control service is in the `VC_SERVICE_STATE_NONE` state:
+
+  ```c
+  void
+  get_current_language()
+  {
+      int ret;
+      char* current_lang = NULL;
+      ret = vc_mgr_get_current_language(&current_lang);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get the microphone volume during recording using `vc_mgr_get_recording_volume()`.
+  The recording volume value is retrieved periodically with the short-term recorded sound data as decibels (dB). The recording volume normally has a negative value, and `0` is the maximum value. `vc_mgr_get_recording_volume()` is used when the voice control service is in the `VC_SERVICE_STATE_RECORDING` state:
+
+  ```c
+  void get_recording_volume()
+  {
+      int ret;
+      float recording_volume = 0.0f;
+      ret = vc_mgr_get_recording_volume(&recording_volume);
+      if (VC_ERROR_NONE != ret)
+          /* Error handling */
+  }
+  ```
+
+- Get or set the audio in type using `vc_mgr_get_audio_type()` and `vc_mgr_set_audio_type()`.
+  The audio type values can be `VC_AUDIO_ID_BLUETOOTH` or `VC_AUDIO_ID_MSF` in the string. These functions are used when the voice control manager is in the `VC_STATE_READY` state:
```

Comment:
```suggestion
- Get or set the audioIn type using `vc_mgr_get_audio_type()` and `vc_mgr_set_audio_type()`.
  The audio type values can be `VC_AUDIO_ID_BLUETOOTH` or `VC_AUDIO_ID_MSF` in the string. These functions are used when the voice control manager is in the `VC_STATE_READY` state:
```

Please verify is it correct to write audioIn?

Reply:
Yes, the type means where the audio source is comming from.
Thus, I think just `audio type` is okay, because it is the same as the function name.